crontabber tests assume a UTC database

In [socorro](https://github.com/mozilla/socorro) we assume a UTC postgres but it's not necessarily something we want to do for everybody using crontabber. 

If you don't have a UTC database you get this:

```
(crontabber)peterbe@mpb:~/dev/PYTHON/crontabber (master=)$ nosetests
.........F..FF..F...................................F...........................
======================================================================
FAIL: test_backfill_job_at_specific_hour (crontabber.tests.test_crontabber.TestCrontabber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/crontabber/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/peterbe/dev/PYTHON/crontabber/crontabber/tests/test_crontabber.py", line 938, in test_backfill_job_at_specific_hour
    information['first_run'].strftime('%H:%M'), '10:30'
AssertionError: '03:30' != '10:30'

======================================================================
FAIL: test_backfilling_with_configured_time_slow_job (crontabber.tests.test_crontabber.TestCrontabber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/crontabber/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/peterbe/dev/PYTHON/crontabber/crontabber/tests/test_crontabber.py", line 1352, in test_backfilling_with_configured_time_slow_job
    '18:00:00'
AssertionError: '11:00:00' != '18:00:00'

======================================================================
FAIL: test_basic_job_at_specific_hour (crontabber.tests.test_crontabber.TestCrontabber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/crontabber/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/peterbe/dev/PYTHON/crontabber/crontabber/tests/test_crontabber.py", line 846, in test_basic_job_at_specific_hour
    information['first_run'].strftime('%H:%M'), '10:30'
AssertionError: '03:30' != '10:30'

======================================================================
FAIL: test_basic_run_job_with_hour (crontabber.tests.test_crontabber.TestCrontabber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/crontabber/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/peterbe/dev/PYTHON/crontabber/crontabber/tests/test_crontabber.py", line 627, in test_basic_run_job_with_hour
    information['next_run'].strftime('%H:%M:%S'), '03:00:00'
AssertionError: '20:00:00' != '03:00:00'

======================================================================
FAIL: test_slow_backfilled_timed_daily_job (crontabber.tests.test_crontabber.TestCrontabber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/crontabber/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/peterbe/dev/PYTHON/crontabber/crontabber/tests/test_crontabber.py", line 1445, in test_slow_backfilled_timed_daily_job
    '10:00:00'
AssertionError: '03:00:00' != '10:00:00'

----------------------------------------------------------------------
Ran 80 tests in 5.884s

FAILED (failures=5)
```

--- a/crontabber/app.py
+++ b/crontabber/app.py
@@ -141,8 +141,6 @@ class JobStateDatabase(RequiredConfig):
             CREATE_CRONTABBER_LOG_SQL
         )
 
-
-
     def has_data(self):
         try:
             return bool(self.transaction_executor(

--- a/crontabber/tests/base.py
+++ b/crontabber/tests/base.py
@@ -125,8 +125,6 @@ class IntegrationTestCaseBase(TestCaseBase):
     def setUp(self):
         super(IntegrationTestCaseBase, self).setUp()
         self.config = self.get_standard_config()
-        # instanciate one of these to make sure the tables are created
-        app.JobStateDatabase(self.config.crontabber)
 
         dsn = (
             'host=%(database_hostname)s '
@@ -142,7 +140,21 @@ class IntegrationTestCaseBase(TestCaseBase):
                 'real database'
             )
         self.conn = psycopg2.connect(dsn)
-        self._truncate()
+        cursor = self.conn.cursor()
+        cursor.execute('SHOW timezone;')
+        try:
+            failed = True
+            tz, = cursor.fetchone()
+            if tz != 'UTC':
+                cursor.execute("""
+                   ALTER DATABASE %(database_name)s SET TIMEZONE TO UTC;
+                """ % self.config.crontabber)
+            failed = False
+        finally:
+            failed and self.conn.rollback() or self.conn.commit()
+
+        # instanciate one of these to make sure the tables are created
+        app.JobStateDatabase(self.config.crontabber)
 
     def _truncate(self):
         self.conn.cursor().execute("""

